ionic2/app/components/map/map.component.ts

```diff
@@ -67,15 +68,22 @@ export class MapComponent {
       this.map.navigate(start, destination);
     })
   }
+}
 
-  private static mapPokePOI(pokePOI: Object): (PokeSighting | PokePrediction | PokeMob) {
-    if ('source' in pokePOI) {
-      return PokeSighting.fromObject(pokePOI);
-    } else if ('clusterId' in pokePOI) {
-      return PokeMob.fromObject(pokePOI);
-    } else {
-      throw new Error('PokePOI cannot be identified as PokeSighting or PokeMob:\n' + JSON.stringify(pokePOI));
-    }
+function poiFromMapEventData(rawData: any) : POI {
+  if ('source' in rawData) {
+    const sighting = new Sighting();
+    sighting.latitude = rawData.location.coordinates[1];
+    sighting.longitude = rawData.location.coordinates[0];
+    sighting.pokemonId = rawData.pokemonId;
+    sighting.source = rawData.source;
+    sighting.appearedOn = rawData.appearedOn;
+    return sighting;
+  } else if ('clusterId' in rawData) {
+    const mob = new Mob();
+    return mob;
```

Comment:
Does the map already produce mobs? If so, what's the exact format of the event data?